.buildkite/hooks/post-checkout

```diff
@@ -32,6 +34,49 @@ checkout_merge() {
     }
 }
 
+retry() {
+    local retries=$1
+    shift
+
+    local count=0
+    until "$@"; do
+        exit=$?
+        wait=$((2 ** count))
+        count=$((count + 1))
+        if [ $count -lt "$retries" ]; then
+            >&2 echo "Retry $count/$retries exited $exit, retrying in $wait seconds..."
+            sleep $wait
+        else
+            >&2 echo "Retry $count/$retries exited $exit, no more retries left."
+            return $exit
+        fi
+    done
+    return 0
+}
+
+GITHUB_REPO_TOKEN=$(retry 5 vault kv get -field token ${GITHUB_TOKEN_VAULT_PATH})
```

Comment:
I placed this part in the post-checkout hook because I need to define the variable `IS_FILE_EXISTS_IN_REPO` before the pipeline initialization (uploading stage). The `rda-snapshot` stage depends on the value of this variable: https://github.com/elastic/fleet-server/blob/4b95d6b1c3e091da204e86df7c8b797a10144495/.buildkite/pipeline.package.mbp.yml#L43

Reply:
I tested it and we can repeat the test together.

Reply:
Thanks for testing it again @sharbuz . Nice to see that some variables can be calculated in the post-checkout hook and then used in the own step definition